tests/ui/mir/alignment/packed.rs

```diff
@@ -0,0 +1,26 @@
+// run-pass
+// compile-flags: -C debug-assertions
+
+#![feature(strict_provenance)]
+
+#[repr(packed)]
+struct Misaligner {
+    _head: u8,
+    tail: u64,
+}
+
+fn main() {
+    let v = Misaligner {
+        _head: 0u8,
+        tail: 42,
+    };
+    // Check that we can take the addr_of! a packed field that is not aligned
+    let ptr = std::ptr::addr_of!(v.tail);
+    assert!(ptr.addr() % std::mem::align_of::<u64>() > 0);
```

Comment:
FWIW this assertion can fail if the address of `v` happens to be just 1 below a multiple of 8.

Reply:
I think I handled this by allocating two of the packed structs in an array.